### PR TITLE
Remove deprecated and unneeded `defaultProps`

### PR DESCRIPTION
**Pull request by Lepozepo**
### Initial checklist

*   [ ] I read the support docs 
*   [ ] I read the contributing guide 
*   [ ] I agree to follow the code of conduct 
*   [ ] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [ ] If applicable, I’ve added docs and tests

### Description of changes
Seeing a deprecation warning with `defaultProps` but for some reason this didn't actually remove it in my environment. Regardless, this is a safer bet

TODO

### CODE REVIEW

**Comment by wooorm** [comment]
- PR info not filled out
- No test
- Build fails
- We already support no `defaultProps`: https://github.com/remarkjs/react-markdown/blob/9b2044012ff1592a78be6b944fd99d34e155b0a8/lib/ast-to-react.js#L237
- “for some reason this didn't actually remove it in my environment” sounds like this doesn’t solve your original question

**Comment by Lepozepo** [comment]
@wooorm ah, yeah, I was just going through it quickly and didn't really look at or test anything. More than anything wanted to bring the warning to light. I'll see if I can make a bit of time tonight to play around with the anchor tag, sounds like maybe I messed something up on my end to trigger the warning on the `a` tag

**Comment by wooorm** [comment]
Maybe better to post the problem you ran into?

**Comment by Lepozepo** [comment]
@wooorm seeing this in a nextjs app

Screenshot 2022-12-13 at 8 26 42 AM

**Comment by actraiser** [comment]
Seeing the same error in nextjs 13

File: lib/react-markdown.js [hunk 1]
```diff
@@ -71,7 +71,8 @@ const deprecated = {
  * @param {ReactMarkdownOptions} options
  * @returns {ReactElement}
  */
-export function ReactMarkdown(options) {
+export function ReactMarkdown(_options) {
+  const options = { ..._options, transformLinkUri: _options?.transformLinkUri || uriTransformer };
```
Comment (wooorm) [inline lib/react-markdown.js:75]:
```suggestion
export function ReactMarkdown(options) {
```

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
Thanks!

**Comment by william-liebenberg** [comment]
hi all, when would this build get released? im getting this exact error :)

**Comment by wooorm** [comment]
Soon. It’s a warning

**Comment by william-liebenberg** [comment]
Thanks @wooorm :)

**Comment by ChristianMurphy** [comment]
It is fixed an released, please update the version of `react-markdown` to latest (8.0.5 at the time of writing)

**Pull request closed without merging**